### DISCUSSION TITLE
Not able to do `./install-foundry-zksync` after cloning the repo.

### DISCUSSION THREAD

**Discussion by Jain-Sameer** [Cyfrin/foundry-full-course-cu#2101] [General]
Whenever I try to run the command :
```
./install-foundry-zksync
```
This is the error I'm getting while doing this :-
```
Archive:  /tmp/tmp.cgnHyUI7Tb/foundry-zksync.zip 
End-of-central-directory signature not found.  Either this file is not a zipfile, or it constitutes one disk of a multi-part archive.  In the latter case the central directory and zipfile comment will be found on the last disk(s) of this archive. 
unzip:  cannot find zipfile directory in one of /tmp/tmp.cgnHyUI7Tb/foundry-zksync.zip or
 /tmp/tmp.cgnHyUI7Tb/foundry-zksync.zip.zip, and cannot find /tmp/tmp.cgnHyUI7Tb/foundry-zksync.zip.ZIP, period.
foundryup-zksync: command failed: unzip /tmp/tmp.cgnHyUI7Tb/foundry-zksync.zip -d /c/Users/XXXXX/.foundry/bin
```
Please help me out here!

**Comment by EngrPips** [comment]
Hey @Jain-Sameer, did you `cd foundry-zksync` before trying to run `./install-foundry-zksync`?

**Reply by Jain-Sameer** [reply]
Yes, I did. I tried once again. Its showing the same error! https://github.com/Cyfrin/foundry-full-course-cu/discussions/2101

**Reply by EngrPips** [reply]
I haven't encountered this error before, so it can be hard to say what is wrong. Can you start the process afresh? Let's see how it goes. Make sure to follow the **QUICK INSTALL** section of this https://github.com/matter-labs/foundry-zksync repo.

**Reply by Jain-Sameer** [reply]
I have done it, multiple times! Still get the same error. 😢

**Reply by Jain-Sameer** [reply]
![Error File](https://github.com/user-attachments/assets/55133e6a-3e13-4cde-a08e-bc63f40cfb4e)

All still the same!

**Reply by EngrPips** [reply]
It seems like you are running on `wsl,` and I have been doing all my processes on Mac. I am unsure what could be wrong here, but let's patiently await @cromewar suggestions whenever they are available.

**Reply by Jain-Sameer** [reply]
okay! Thanks for taking the time though!

**Reply by EngrPips** [reply]
You are welcome. You should also ask an AI buddy, as that might help.

**Reply by ighodaro-daniel** [reply]
do you have enough space on your disk

**Reply by Jain-Sameer** [reply]
yes! more than 150 gigs free. ChatGPT said the same.

**Comment by ighodaro-daniel** [comment]
do you have enough space on your disk

**Reply by Jain-Sameer** [reply]
yes! more than 150 gigs free. ChatGPT said the same.

**Comment by ronijaat** [comment]
I am also facing the same issue.

**Reply by EngrPips** [reply]
Hello @ronijaat, Please check if anyone has reported this same issue on the official Github of repo of `foundry-zksync` `https://github.com/matter-labs/foundry-zksync`. If not, you can open an issue by providing details about the issue, and maybe they can help with some solution.

**Reply by Jain-Sameer** [reply]
> I am also facing the same issue.

If you find any solution, pls let me know!

**Reply by ronijaat** [reply]
> Hello @ronijaat, Please check if anyone has reported this same issue on the official Github of repo of `foundry-zksync` `https://github.com/matter-labs/foundry-zksync`. If not, you can open an issue by providing details about the issue, and maybe they can help with some solution.

I found one that was opened 3 weeks ago, but there is no solution for it yet.

**Reply by EngrPips** [reply]
I will comment on the issue to at least draw attention to the issue.

**Reply by ronijaat** [reply]
https://github.com/matter-labs/foundry-zksync/issues/472#issue-2404913920

**Reply by EngrPips** [reply]
Good. Let's hope for a response. @PatrickAlphaC want to call your attention to the fact that people using `wsl` are having issues installing `foundry-zksync`

**Reply by PatrickAlphaC** [reply]
I have pinged the team, and they are just a bit swamped right now. 

For all those having issues installing the zksync fork of foundry, just skip installing it and work with foundry. Treat any ZKSync section as optional. Once the tool is in a better place, you'll be able to do ZKSync stuff no problem.

**Reply by EngrPips** [reply]
Thanks for the response, sir.

**Comment by Aashim-Limbu** [comment]
Any updates in this issue ? Still having the issue with gitbash.
```bash
Archive:  /tmp/tmp.JhNuYfBp9v/foundry-zksync.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of /tmp/tmp.JhNuYfBp9v/foundry-zksync.zip or
        /tmp/tmp.JhNuYfBp9v/foundry-zksync.zip.zip, and cannot find /tmp/tmp.JhNuYfBp9v/foundry-zksync.zip.ZIP, period.
foundryup-zksync: command failed: unzip /tmp/tmp.JhNuYfBp9v/foundry-zksync.zip -d /c/Users/user/.foundry/bin
```

**Reply by EngrPips** [reply]
I am unsure what the update is. @PatrickAlphaC  any help will be appreciated sir.

**Reply by BlackDranzer777** [reply]
**A naming conflict occurs when a directory named foundryup-zksync exists in the same location where the installation script attempts to download a file with the same name.**

1. Move the Conflicting Directory

> Run the following command in your WSL:
> `mv foundryup-zksync foundryup-zksync-backup`
> 
> Delete it:
> `rm -rf foundryup-zksync`

2. Re-run the Installation Script:

> `bash install-foundry-zksync`

**Reply by EngrPips** [reply]
Thanks for the heads-up @BlackDranzer777

**Reply by BlackDranzer777** [reply]
Let me know if this resolves the issue @Jain-Sameer.

**Reply by Jain-Sameer** [reply]
Nah, it doesn't.

**Comment by WhiteKnightK** [comment]
I'm using the Windows system and have encountered the same issue. After trying for three days and consulting various documents, I still couldn't solve it. I suggest that friends who later encounter this problem skip this part first, especially if you are also using the Windows system.

---

### Solution:

Subsequently, I managed to download it successfully using WSL2. If other methods don't work for you, you can give WSL2 a try.

**Reply by EngrPips** [reply]
Thats tough.

**Reply by KMean** [reply]
weird for me is working all fine, also cloning and installing manually. 
```bash
git clone https://github.com/matter-labs/foundry-zksync.git
Cloning into 'foundry-zksync'...
remote: Enumerating objects: 85876, done.
remote: Counting objects: 100% (110/110), done.
remote: Compressing objects: 100% (83/83), done.
remote: Total 85876 (delta 58), reused 51 (delta 26), pack-reused 85766 (from 1)
Receiving objects: 100% (85876/85876), 48.03 MiB | 1.70 MiB/s, done.
Resolving deltas: 100% (63112/63112), done.
```
```bash
./foundry-zksync/install-foundry-zksync
Downloading the install script...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2272  100  2272    0     0  12062      0 --:--:-- --:--:-- --:--:-- 12085
Making install script executable...
Running the installation script...
Installing foundryup-zksync...

Detected your preferred shell is bash and added foundryup-zksync to PATH.
Run 'source /home/kmean/.bashrc' or start a new terminal session to use foundryup-zksync.
Then, simply run 'foundryup-zksync' to install foundry-zksync.
Sourcing the shell configuration file: '/home/kmean/.bashrc'
Downloading foundryup-zksync...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13002  100 13002    0     0  61944      0 --:--:-- --:--:-- --:--:-- 61914
Making foundryup-zksync executable...
Running foundryup-zksync setup...


.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

 ╔═╗ ╔═╗ ╦ ╦ ╔╗╔ ╔╦╗ ╦═╗ ╦ ╦         Portable and modular toolkit
 ╠╣  ║ ║ ║ ║ ║║║  ║║ ╠╦╝ ╚╦╝    for ZKsync Application Development
 ╚   ╚═╝ ╚═╝ ╝╚╝ ═╩╝ ╩╚═  ╩                 written in Rust.

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

Fork of    : https://github.com/foundry-rs/
Repo       : https://github.com/matter-labs/foundry-zksync/
Book       : https://foundry-book.zksync.io/

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

foundryup-zksync: downloading latest forge, and cast
######################################################################################################################################################################################################################################################### 100.0%
foundryup-zksync: attempting to download manpages from release assets
foundryup-zksync: Found manpage tarball: https://github.com/matter-labs/foundry-zksync/releases/download/foundry-zksync-v0.0.4/foundry_man_nightly-ae913af65381734ad46c044a9495b67310bc77c4.tar.gz
######################################################################################################################################################################################################################################################### 100.0%
foundryup-zksync: installed - forge 0.0.4 (VERGEN_IDEMPOTENT_OUTPUT 2025-01-23T19:54:33.239256966Z)
foundryup-zksync: installed - cast 0.0.4 (VERGEN_IDEMPOTENT_OUTPUT 2025-01-23T19:54:33.227020357Z)
foundryup-zksync: downloading latest anvil-zksync
######################################################################################################################################################################################################################################################### 100.0%
foundryup-zksync: installed - anvil-zksync 0.2.5
foundryup-zksync: done!
Cleaning up installation artifacts...
Cleanup completed.
Installation completed successfully!
Verifying installation...
forge 0.0.4 (VERGEN_IDEMPOTENT_OUTPUT 2025-01-23T19:54:33.239256966Z)
Forge is successfully installed with version: forge 0.0.4 (VERGEN_IDEMPOTENT_OUTPUT 2025-01-23T19:54:33.239256966Z)
```


I'm on windows running wsl2 so I still prefer installing it with
```bash
curl -L https://raw.githubusercontent.com/matter-labs/foundry-zksync/main/install-foundry-zksync | bash
```
and then 
```bash
foundryup-zksync
```
as described on the [installation page](https://foundry-book.zksync.io/getting-started/installation)

**Reply by EngrPips** [reply]
Interesting.

**Reply by WhiteKnightK** [reply]
I tried your method, but strangely enough, I still got stuck at the step of ``./foundry-zksync/install-foundry-zksync`` and kept encountering the same errors as @Jain-Sameer did.

**Reply by WhiteKnightK** [reply]
> 这很难。

Yes, it really bothers me. I have skipped this part and started to learn the following content.

**Reply by KMean** [reply]
> I tried your method, but strangely enough, I still got stuck at the step of `./foundry-zksync/install-foundry-zksync` and kept encountering the same errors as @Jain-Sameer did.

Did you try wsl2?

**Reply by EngrPips** [reply]
> > 这很难。
> 
> Yes, it really bothers me. I have skipped this part and started to learn the following content.

Tough

**Reply by PatrickAlphaC** [reply]
Hi @WhiteKnightK, you'll save yourself a lot of headache if you get WSL now :)

Windows is _really_ rough for smart contract development work, you'll likely keep getting errors. Could you try to install it and go from there?

**Reply by EngrPips** [reply]
Thanks for the heads-up sir

**Reply by WhiteKnightK** [reply]
> > I tried your method, but strangely enough, I still got stuck at the step of `./foundry-zksync/install-foundry-zksync` and kept encountering the same errors as @Jain-Sameer did.
> 
> Did you try wsl2?

Thank you, sir. I successfully installed Foundry-Zksync using WSL2.

**Reply by WhiteKnightK** [reply]
> Hi @WhiteKnightK, you'll save yourself a lot of headache if you get WSL now :)
> 
> Windows is _really_ rough for smart contract development work, you'll likely keep getting errors. Could you try to install it and go from there?

Thank you, sir. I successfully installed Foundry-Zksync using WSL2.

**Reply by WhiteKnightK** [reply]
> Thanks for the heads-up sir

I retract my previous statement. Maybe you can give WSL2 a try. I managed to install it successfully today using it. However, it's still quite troublesome. It took me several hours to get it done.

**Reply by EngrPips** [reply]
> > Thanks for the heads-up sir
> 
> I retract my previous statement. Maybe you can give WSL2 a try. I managed to install it successfully today using it. However, it's still quite troublesome. It took me several hours to get it done.

Yeah working on windows for smart contract development can be kind of tough.

**Reply by PatrickAlphaC** [reply]
@WhiteKnightK the hardest part of a lot of development is installing the tools. Glad you made it through the toughest hurdle! Life gets easier now!

**Reply by sivaren** [reply]
I'm a windows user and facing the same issue if I try the installation with `BASH Terminal` both for `curl -L https://raw.githubusercontent.com/matter-labs/foundry-zksync/main/install-foundry-zksync | bash` command and manual way with cloning the `foundry-zksync` repo. 

My **conclusion** goes to `WSL Terminal` is the only solution for windows user to use **foundry-zksync** for current development.

**Reply by KMean** [reply]
> I'm a windows user and facing the same issue if I try the installation with `BASH Terminal` both for `curl -L https://raw.githubusercontent.com/matter-labs/foundry-zksync/main/install-foundry-zksync | bash` command and manual way with cloning the `foundry-zksync` repo. 
> 
> 
> 
> My **conclusion** goes to `WSL Terminal` is the only solution for windows user to use **foundry-zksync** for current development.

Absolutely! And is also written in the documentation 

![image](https://github.com/user-attachments/assets/0d460025-2eea-4b09-b690-e663f03b668d)

**Reply by EngrPips** [reply]
Interesting

**Comment by Alatharee21** [comment]
2026 and this problem still exist

**Reply by EngrPips** [reply]
The tool docs should be sufficient to guide one through proper installation.
